server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -201,13 +202,16 @@ public Stream<Reference> getReferences() {
 
       // combine all the entries read from file and scan columns in the metadata table
       Stream<StoredTabletFile> fileStream = tm.getFiles().stream();
+      // map the files to Reference objects
+      var stream = fileStream.map(f -> new ReferenceFile(tm.getTableId(), f.getMetaUpdateDelete()));
+
       // scans are normally empty, so only introduce a layer of indirection when needed
       final var tmScans = tm.getScans();
       if (!tmScans.isEmpty()) {
-        fileStream = Stream.concat(fileStream, tmScans.stream());
+        var scanStream =
+            tmScans.stream().map(s -> new ReferenceScan(tm.getTableId(), s.getMetaUpdateDelete()));
```

Comment:
Could just have a different constructor for these. Or, use static methods to get "named" constructors, as in:

```java
  ReferenceFile fileRef = ReferenceFile.forRFile(tableId, f.getMetaUpdateDelete());
  ReferenceFile scanRef = ReferenceFile.forScan(tableId, s.getMetaUpdateDelete());
```

That way you don't have to deal with the equals/hashCode issues, and can just use the isScan() method. Each of these static methods would call a private internal constructor:

```java
  private ReferenceFile(TableId tableId, String metadataEntry, boolean isScan);
```

(can make the constructor package-private if you need it visible to be mocked for a unit test)

Reply:
I like making RefrenceFile's constructor take a boolean isScan parameter instead of creating a new subclass. The static methods mentioned seem like a nice way to do that.

Reply:
I like the static methods for "named" constructors. 
I've gone ahead and implemented them for use instead of ReferenceScan. 

It looks like we could also use them to replace the ReferenceDirectory class as well, but I'd want to handle that in a separate PR.

Reply:
@ddanielr To close the loop on this thread, did you end up creating a separate PR for this?

Reply:
PR #4724 was created to close the loop